frappe/utils/change_log.py

```diff
@@ -124,3 +126,84 @@ def get_app_last_commit_ref(app):
 			shell=True).strip()[:7]
 	except Exception as e:
 		return ''
+
+def check_for_update():
+	updates = frappe._dict(major=[], minor=[], patch=[])
+	apps    = get_versions()
+
+	for app in apps:
+		app_details = check_release_on_github(app)
+		if not app_details: continue
+
+		github_version, org_name = app_details
+		# Get local instance's current version or the app
+		instance_version = Version(apps[app]['version'])
+		# Compare and popup update message
+		for update_type in updates:
+			if github_version.__dict__[update_type] > instance_version.__dict__[update_type]:
+				updates[update_type].append(frappe._dict(
+					current_version   = str(instance_version),
+					available_version = str(github_version),
+					org_name          = org_name,
+					app_name          = app,
+					title             = apps[app]['title'],
+				))
+				break
+
+	generate_update_message(updates)
+
+def check_release_on_github(app):
+		# Check if repo remote is on github
+	remote_url = subprocess.check_output("cd ../apps/{} && git ls-remote --get-url".format(app), shell=True)
+	if "github.com" not in remote_url:
+		return None
+
+	# Get latest version from github
+	if 'https' not in remote_url:
+		return None
+
+	org_name = remote_url.split('/')[3]
+	r = requests.get('https://api.github.com/repos/{}/{}/releases'.format(org_name, app))
+	if r.status_code == 200 and r.json():
+		# 0 => latest release
+		return Version(r.json()[0]['tag_name'].strip('v')), org_name
+	else:
+		# In case of an improper response or if there are no releases
+		return None
+
+def generate_update_message(updates):
+	update_message = ""
+	for update_type in updates:
+		release_links = ""
+		for app in updates[update_type]:
+			release_links += "<a href='https://github.com/{org_name}/{app_name}/releases/tag/v{available_version}'><b>{title}</b>: v{available_version}</a><br>".format(
+				available_version = app.available_version,
+				org_name          = app.org_name,
+				app_name          = app.app_name,
+				title             = app.title
+			)
+		if release_links:
+			update_message += _("New {} releases for the following apps are available") + ":<br><br>{}<hr>".format(update_type, release_links)
```

Comment:
Not a good idea to encode the translated message (if the users have different languages)?

Better to just save the release JSONs and build the message at the point of display.